homeassistant/helpers/event.py

```diff
@@ -70,6 +72,7 @@ def track_point_in_utc_time(hass, action, point_in_time):
     point_in_time = dt_util.as_utc(point_in_time)
 
     @ft.wraps(action)
+    @asyncio.coroutine
```

Comment:
I think that these are actually the biggest time improvement here.

We fire a `time_changed` event every second with the current time stamp. Each point in time listener will check if it's time is already due and fire if necessary. That means 3600 checks for something planned into an hour. Running this on the loop should speed things up greatly. I don't want to migrate to `loop.call_at()` just yet, because it means that we're stepping away from being a fully event driven system.